src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+
+        if (block.timestamp >= cycleExpiryDate) revert CycleHasExpired();
+
+        // Get payer address and amount paid from JB DidPay data struct
+        address payer = _data.payer;
+        JBTokenAmount memory amount = _data.amount;
+        uint256 paymentAmount = amount.value;
+
+        // Update storage variables
+        totalAmountPledged += paymentAmount;
+        pledgedAmount[payer] = paymentAmount;
+        pledgers.push(payer);
+        orderToPledgerToAmount[pledgeOrder][payer] = paymentAmount;
+        pledgeOrder++;
+    }
+
+    /**
+     * @notice After the funding cycle ends, anyone can call this function to store the cycle results.
+     * @dev
+     */
+    function relayCycleResults() public {
+        if (isCycleExpired == true) revert FunctionHasAlreadyBeenCalled();
+        if (block.timestamp < cycleExpiryDate) revert CycleHasNotEndedYet(cycleExpiryDate);
+        // This function can only be called once.
+        isCycleExpired = true;
+
+        // Sanity check comparison with JB total amount pledged. If same, lock in. If different,
+        // go to fallback scenario.
+        // totalAmountPledged_fromJB = paymentTerminalStore.balanceOf(paymentTerminal, projectId);
+        // if (totalAmountPledged != totalAmountPledged_fromJB) {
+        //     totalAmountPledged = totalAmountPledged_fromJB;
+        // Need fallback scenario here, or is ^ this adequate?
+        // }
+        if (totalAmountPledged >= cycleTarget) {
+            isTargetMet = true;
+        }
+
+        emit CycleHasClosed(isCycleExpired, isTargetMet);
+    }
+
+    /// @notice This function gets called when the project's token holders redeem.
+    /// @dev Part of IJBFundingCycleDataSource.
+    /// @param _data Standard Juicebox project redemption data. See https://docs.juicebox.money/dev/api/data-structures/jbredeemparamsdata/.
+    /// @return reclaimAmount Amount to be reclaimed from the treasury. This is useful for optionally customizing how much funds from the treasury are disbursed per redemption.
+    /// @return memo A memo to be forwarded to the event. Useful for describing any new actions are being taken.
+    /// @return delegateAllocations Amount to be sent to delegates instead of being added to the beneficiary. Useful for auto-routing funds from a treasury as redemptions are sought.
+    function redeemParams(JBRedeemParamsData calldata _data)
+        external
+        view
+        virtual
+        override
+        returns (
+            uint256 reclaimAmount,
+            string memory memo,
+            JBRedemptionDelegateAllocation3_1_1[] memory delegateAllocations
+        )
+    {
+        address payable redeemer = payable(_data.holder);
+
+        // The if statements are the only changes from the redeemParams() template.
+        if (pledgedAmount[redeemer] == 0) revert MustBePledger();
+        if (isCycleExpired != true) revert CycleHasNotEndedYet(cycleExpiryDate);
```

Comment:
would just use `if (!isCycleExpired)` for bools. No need to check explicitly against true

Reply:
Coo cool. Will do.